packages/components/src/components/post-collapsible/post-collapsible.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Element, h, Method, Prop, State } from '@stencil/core';
+import { checkValidHTMLTag, getElementHeight, onTransitionEnd } from '../../utils';
+
+let nextId = 0;
+
+@Component({
+  tag: 'post-collapsible',
+  styleUrl: 'post-collapsible.scss',
+  shadow: true,
+})
+export class PostCollapsible {
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop({ mutable: true }) collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the collapsible header within the headings structure.
+   */
+  @Prop() headingLevel?: number = 2;
+
+  @State() collapseClasses: string;
+  @State() collapseHeight: string | null = null;
+
+  @Element() host: HTMLElement;
+
+  hasHeader: boolean;
+  headingTag: string | undefined;
+  collapsibleId: string;
+  collapsibleElement: HTMLElement;
+
+  componentWillLoad() {
+    this.hasHeader = !!this.host.querySelector('[slot="header"]');
+    if (!this.hasHeader) {
+      console.warn('Be sure to bind the post-collapsible to its control using aria-controls and aria-expanded attributes. More information here: https://getbootstrap.com/docs/5.2/components/collapse/#accessibility');
+    }
+
+    this.headingTag = checkValidHTMLTag(`h${this.headingLevel}`, 'The post-collapsible element requires a heading level between 1 and 6.');
+
+    this.collapsibleId = this.host.id || `post-collapsible-${nextId++}`;
+    this.collapseClasses = this.collapsed ? 'collapse' : 'collapse show';
+  }
+
+  componentDidLoad() {
+    this.collapsibleElement = this.host.shadowRoot.querySelector(`#${this.collapsibleId}--collapse`);
+  }
+
+  /**
+   * Triggers the collapse programmatically.
+   */
+  @Method()
+  async toggle(open: boolean = this.collapsed) {
+    if (open === this.collapsed) {
+      this.collapsed = !open;
+
+      this.startTransition();
+
+      await onTransitionEnd(this.collapsibleElement).then(() => {
+        this.collapseHeight = null;
+        this.collapseClasses = this.collapsed ? 'collapse' : 'collapse show';
+      });
+    }
+  }
+
+  startTransition() {
+    const expandedHeight = getElementHeight(this.collapsibleElement, 'show');
+
+    this.collapseHeight = this.collapsed ? expandedHeight : '0px';
+    this.collapseClasses = 'collapsing';
+
+    setTimeout(() => {
+      this.collapseHeight = this.collapsed ? '0px' : expandedHeight;
+    });
```

Comment:
If the util "getElementHeight" returns a number (without 'px'):

```suggestion
    const expandedHeight = getElementHeight(this.collapsibleElement, 'show');

    this.collapseHeight = `${this.collapsed ? expandedHeight : 0}px`;
    this.collapseClasses = 'collapsing';

    setTimeout(() => {
      this.collapseHeight = `${this.collapsed ? 0 : expandedHeight}px`;
    });
```